tests/unit/pages/offer-details/OfferDetails.spec.jsx

```diff
@@ -102,37 +105,45 @@ describe('OfferDetails on desktop', () => {
 
     await waitFor(() => expect(confirmationText).not.toBeInTheDocument())
   })
+})
 
-  it('should open modal window with enroll offer', async () => {
-    const newMockState = {
-      appMain: { userId: '6421d9833cdf38b706756dff', userRole: 'student' }
-    }
+describe('Offer details with student role', () => {
+  beforeEach(() => {
+    useBreakpoints.mockImplementation(() => desktopData)
 
     renderWithProviders(<OfferDetails />, {
-      preloadedState: newMockState
+      preloadedState: {
+        appMain: { userId: '6421d9833cdf38b706756dff', userRole: 'student' }
+      }
     })
 
+    mockAxiosClient
+      .onGet(`${URLs.offers.get}/${mockOffer._id}`)
+      .reply(200, mockOffer)
+    mockAxiosClient
+      .onGet(`${URLs.categories.get}${URLs.subjects.get}${URLs.offers.get}`)
+      .reply(200, { offers: [], count: 0 })
+  })
+
+  it('should open modal window with enroll offer', async () => {
     const enrollOffer = await screen.findByText('common.labels.enrollOffer')
 
     fireEvent.click(enrollOffer)
 
-    const modalTitle = await screen.findByText(
-      'offerDetailsPage.enrollOffer.title'
-    )
+    const modalTitle = screen.getByText('offerDetailsPage.enrollOffer.title')
 
     expect(modalTitle).toBeInTheDocument()
   })
 })
 
 describe('OfferDetails on mobile', () => {
-  const desktopData = {
+  const mobileData = {
```

Comment:
You moved desktopData to the top, so you should do the same with mobileData.

Reply:
Not really, because _mobileData_ is only used once at one describe block.